music_assistant/server/providers/deezer/__init__.py

```diff
@@ -518,16 +533,46 @@ def parse_playlist(self, playlist: deezer.Playlist) -> Playlist:
             metadata=MediaItemMetadata(
                 images=[MediaItemImage(type=ImageType.THUMB, path=playlist.picture_big)],
             ),
-            is_editable=playlist.creator.id == self.client.user.id,
+            is_editable=self.get_playlist_creator(playlist).id == self.user.id,
         )
 
-    def parse_track(
+    def get_playlist_creator(self, playlist: deezer.Playlist):
+        """See https://twitter.com/Un10cked/status/1682709413889540097."""
+        try:
+            return playlist.creator
+        except AttributeError:
+            return playlist.user
+
+    async def parse_track(
         self,
         track: deezer.Track,
         user_country: str,
         extra_init_kwargs: dict[str, Any] | None = None,
-    ) -> Track | PlaylistTrack:
+    ) -> Track | PlaylistTrack | AlbumTrack:
         """Parse the deezer-python track to a MASS track."""
+        try:
+            deezer_artist = await track.get_artist()
+            artist = ItemMapping(
+                MediaType.ARTIST,
+                str(deezer_artist.id),
+                self.instance_id,
+                deezer_artist.name,
+            )
+        except deezer.exceptions.DeezerErrorResponse:
+            artist = ItemMapping(
+                MediaType.ARTIST,
+                self.instance_id,
+            )
+        try:
+            deezer_album = await track.get_album()
+            album = ItemMapping(
```

Comment:
only return ItemMapping here if you get some limited object here.
If you actually get back a Album object here with some details filled in already (like image etc), just use that.

Reply:
I ended up not fetching either artist or album since its slow